wallet.go

```diff
@@ -301,43 +350,63 @@ func (w *Wallet) SignDigest(digest common.Hash, optChainID ...*big.Int) ([]byte,
 		return nil, nil, fmt.Errorf("SignDigest, subDigestOf: %w", err)
 	}
 
-	sig := &Signature{
-		Threshold: w.config.Threshold,
-		Signers:   SignatureParts{},
-	}
-
-	for _, signerInfo := range w.config.Signers {
-		signer, _ := w.GetSigner(signerInfo.Address)
+	sign := func(ctx context.Context, signerAddress common.Address) (core.SignerSignatureType, []byte, error) {
+		signer, _ := w.GetSigner(signerAddress)
 
 		if signer == nil {
 			// signer isn't available, just include the config value of address
 			// without it's signature
-			sig.Signers = append(sig.Signers, &SignaturePart{
-				Type: SignaturePartTypeAddress, Weight: signerInfo.Weight, Address: signerInfo.Address,
-			})
-			continue
+			// todo: what shall we do in that case? before we returned signer address and weight without signature
+			return core.SignerSignatureTypeEthSign, []byte{}, nil
 		}
 
-		// TODO: in future add support for Sequence Signers, right now we only support EOA
-		// signers in go-sequence
+		if eoaSigner, ok := signer.(MessageSigner); ok {
+			// todo: make sure that this is correct
+			sigValue, err := eoaSigner.SignMessage(subDigest)
+			if err != nil {
+				return core.SignerSignatureTypeEthSign, nil, fmt.Errorf("signer.SignMessage subDigest: %w", err)
+			}
+
+			return core.SignerSignatureTypeEthSign, sigValue, nil
+		} else if seqSigner, ok := signer.(DigestSigner); ok {
+			// todo: make sure that this is correct
+			_, seqSign, err := seqSigner.SignDigest(ctx, common.BytesToHash(subDigest), chainID)
```

Comment:
I think that this should be passing `digest` instead of `subDigest` because if this calls `SignDigest(subDigest)` on the `Wallet` object again, then this line is going to be computing the subdigest of a subdigest: https://github.com/0xsequence/go-sequence/blob/779e241538aec7c657adb4111ab82dc1a8239942/wallet.go#L348.